res/cmake/highfive.cmake

```diff
@@ -13,12 +13,8 @@ if(HighFive)
   endif()
 
   if(DEFINED HDF5_ENABLE_PARALLEL AND "${HDF5_ENABLE_PARALLEL}" STREQUAL "ON")
+    # This if seems to be needed if HDF5 is built from source.
```

Comment:
if we do that we trust the user to have been building HDF5 in parallel indeed, no? What happens if I wrongly build hdf5 from source and forget to enable parallel? You set this variable that sends an inconsistent message to the user that HDF5 parallel is detected whereas it's not and will fail finding parallel function, no?